lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
+        return this._get('order.trades', { 'order_id': order_id }, null, formatResponse);
     };
 
-    /**
-     * Fetch required margin for order/list of orders
-     * @method orderMargins
-     * @memberOf KiteConnect
-     * @instance
-     * @param {Object[]} orders   Margin fetch orders.
-     * @param {string} orders[].exchange    Name of the exchange(eg. NSE, BSE, NFO, CDS, MCX)
-     * @param {string} orders[].tradingsymbol   Trading symbol of the instrument
-     * @param {string} orders[].transaction_type   eg. BUY, SELL
-     * @param {string} orders[].variety   Order variety (regular, amo, bo, co etc.)
-     * @param {string} orders[].product   Margin product to use for the order
-     * @param {string} orders[].order_type   Order type (MARKET, LIMIT etc.)
-     * @param {number} orders[].quantity   Quantity of the order
-     * @param {number} orders[].price   Price at which the order is going to be placed (LIMIT orders)
-     * @param {number} orders[].trigger_price   Trigger price (for SL, SL-M, CO orders)
-     * @param {string} mode    (optional) Compact mode will only give the total margins
-     */
-    self.orderMargins = function(orders, mode=null) {
-        return _post("order.margins", orders, null, null, true,
-                    {"mode":mode});
+    orderMargins(orders: Order[], mode = null): Promise<AxiosResponse<any>> {
+        return this._post('order.margins', orders, null, null, true,
+            { 'mode': mode });
     }
 
-    /**
-     * Fetch basket margin for list of orders
-     * @method orderBasketMargins
-     * @memberOf KiteConnect
-     * @instance
-     * @param {Object[]} orders   Margin fetch orders.
-     * @param {string} orders[].exchange    Name of the exchange(eg. NSE, BSE, NFO, CDS, MCX)
-     * @param {string} orders[].tradingsymbol   Trading symbol of the instrument
-     * @param {string} orders[].transaction_type   eg. BUY, SELL
-     * @param {string} orders[].variety   Order variety (regular, amo, bo, co etc.)
-     * @param {string} orders[].product   Margin product to use for the order
-     * @param {string} orders[].order_type   Order type (MARKET, LIMIT etc.)
-     * @param {number} orders[].quantity   Quantity of the order
-     * @param {number} orders[].price   Price at which the order is going to be placed (LIMIT orders)
-     * @param {number} orders[].trigger_price   Trigger price (for SL, SL-M, CO orders)
-     * @param {string} consider_positions    Boolean to consider users positions while calculating margins
-     * @param {string} mode    (optional) Compact mode will only give the total margins
-     */
-    self.orderBasketMargins = function(orders, consider_positions=true, mode=null) {
-        return _post("order.margins.basket", orders, null, null, true,
-                    {"consider_positions":consider_positions, "mode":mode});
+    orderBasketMargins(orders: Order[], consider_positions = true, mode = null): Promise<AxiosResponse<any>> {
```

Comment:
Missing types for orders, consider_positions and mode. Suggestion:
`orderBasketMargins(orders:Record<string, any>, consider_positions:boolean=true, mode:string="")`